torch/nn/functional.py

```diff
@@ -521,35 +523,38 @@ def adaptive_max_pool3d(input, output_size, return_indices=False):
     output_size: the target output size (single integer)
 """)
 
-adaptive_avg_pool2d = _add_docstr(torch._C._nn.adaptive_avg_pool2d, r"""
-adaptive_avg_pool2d(input, output_size) -> Tensor
 
-Applies a 2D adaptive average pooling over an input signal composed of
-several input planes.
+def adaptive_avg_pool2d(input, output_size):
+    r"""
+    Applies a 2D adaptive average pooling over an input signal composed of
+    several input planes.
 
-See :class:`~torch.nn.AdaptiveAvgPool2d` for details and output shape.
+    See :class:`~torch.nn.AdaptiveAvgPool2d` for details and output shape.
 
-Args:
-    output_size: the target output size (single integer or
-        double-integer tuple)
-""")
+    Args:
+        output_size: the target output size (single integer or
+            double-integer tuple)
+    """
+    output_size = set_output_size(input.size()[2:], output_size)
+    return torch._C._nn.adaptive_avg_pool2d(input, output_size)
 
-adaptive_avg_pool3d = _add_docstr(torch._C._nn.adaptive_avg_pool3d, r"""
-adaptive_avg_pool3d(input, output_size) -> Tensor
 
-Applies a 3D adaptive average pooling over an input signal composed of
-several input planes.
+def adaptive_avg_pool3d(input, output_size):
+    r"""
+    Applies a 3D adaptive average pooling over an input signal composed of
+    several input planes.
 
-See :class:`~torch.nn.AdaptiveAvgPool3d` for details and output shape.
+    See :class:`~torch.nn.AdaptiveAvgPool3d` for details and output shape.
 
-Args:
-    output_size: the target output size (single integer or
-        triple-integer tuple)
-""")
+    Args:
+        output_size: the target output size (single integer or
+            triple-integer tuple)
+    """
+    output_size = set_output_size(input.size()[2:], output_size)
```

Comment:
This will be problematic if input has less than `len(output_size) - 2` dimensions. (loop in `set_output_size` will throw).

How about writing such a helper?
```py
# uses `defaults[-len(input):]` as default values for `input`
def _list_with_default(input, defaults):
  if len(defaults) < len(input):
     raise
  return [v if v is not None else d for v, d in zip(input, default[-len(input):])]
```
You can tweak the name/imple/error message to make it nicer. What do you think?

With something like this you should also add 1 or 2 test that asserts some inputs triggers error.